src/pymor/analyticalproblems/functions.py

```diff
@@ -52,7 +52,7 @@ def __call__(self, x, mu=None):
         """Shorthand for :meth:`~Function.evaluate`."""
         return self.evaluate(x, mu)
 
-    def __add__(self, other):
+    def _add_sub(self, other, sign):
```

Comment:
Maybe this is nitpicking, but your implementation still behaves differently than the one for operators: When `self` is a `Function` that is not an instance of `LincombFunction` but has the `name` `LincombFunction` you will get a different code path. I'm not saying that this is the wrong behavior, but it would be good if it was exactly the same for `Operators`, `Functions` and `ParameterFunctionals`.

In any case, maybe it would make sense to move the logic to some mixin that can be shared by all three types of objects.

Reply:
> Maybe this is nitpicking, but your implementation still behaves differently than the one for operators: When `self` is a `Function` that is not an instance of `LincombFunction` but has the `name` `LincombFunction` you will get a different code path. I'm not saying that this is the wrong behavior, but it would be good if it was exactly the same for `Operators`, `Functions` and `ParameterFunctionals`.

TBH, I do not see the difference code path that you are pointing at. What is the difference between lines 227-238 in constructions.py and lines 65-76 here in functions.py? 

> In any case, maybe it would make sense to move the logic to some mixin that can be shared by all three types of objects.

Could make sense.

Reply:
> TBH, I do not see the difference code path that you are pointing at. What is the difference between lines 227-238 in constructions.py and lines 65-76 here in functions.py?

The difference is that for `Operator` the behavior is implemented in `LincombOperator`, not in `Operator`.

Reply:
True, sorry.
So then, I would actually prefer to move the functionality to `Operator` and fix the fact that an `Operator` which has the `name` `LincombOperator` but actually is not an `LincombOperator` instance is not treated as one.
At this point, I think a mixin is the best way to go.